src/Comparison.dfy

```diff
@@ -0,0 +1,93 @@
+// RUN: %dafny /compile:0 "%s"
+
+/*******************************************************************************
+*  Copyright by the contributors to the Dafny Project
+*  SPDX-License-Identifier: MIT 
+*******************************************************************************/
+
+include "Wrappers.dfy"
+include "Relations.dfy"
+module Comparison {
+
+import opened Wrappers
+import opened Relations
+
+  datatype Cmp = Lt | Eq | Gt {
+    function method Flip(): Cmp {
+      match this
+        case Lt => Gt
+        case Eq => Eq
+        case Gt => Lt
+    }
+
+    const Le? := this != Gt
+    const Ge? := this != Lt
```

Comment:
I definitely like this use of `?` at least, since it's a very natural extension of the built-in `Lt?/Eq?/Gt?` discriminator syntax.